src/flinkStatementResultsManager.test.ts

```diff
@@ -1,137 +1,88 @@
+import { ObservableScope, Scope, Signal } from "inertial";
 import sinon from "sinon";
 import * as messageUtils from "../src/documentProviders/message";
+import {
+  FlinkStatementResultsManagerTestContext,
+  createTestResultsManagerContext,
+} from "../tests/createResultsManager";
 import { eventually } from "../tests/eventually";
 import { loadFixture } from "../tests/fixtures/utils";
 import { createResponseError } from "../tests/unit/testUtils";
-import { StatementResultsSqlV1Api, StatementsSqlV1Api } from "./clients/flinkSql";
-import { FlinkStatementResultsManager } from "./flinkStatementResultsManager";
-import { CCloudResourceLoader } from "./loaders/ccloudResourceLoader";
+import { MessageType, PostFunction } from "./flinkStatementResultsManager";
 import { FlinkStatement, Phase } from "./models/flinkStatement";
-import * as sidecar from "./sidecar";
-import { DEFAULT_RESULTS_LIMIT } from "./utils/flinkStatementResults";
+import { WebviewStorage } from "./webview/comms/comms";
+import {
+  FlinkStatementResultsViewModel,
+  ResultsViewerStorageState,
+} from "./webview/flink-statement-results";
 
-describe("FlinkStatementResultsManager", () => {
+class FakeWebviewStorage<T> implements WebviewStorage<T> {
+  private storage: T | undefined;
+
+  get(): T | undefined {
+    return this.storage;
+  }
+
+  set(state: T): void {
+    this.storage = state;
+  }
+}
+
+function createFakeWebviewStorage<T>(): WebviewStorage<T> {
+  return new FakeWebviewStorage<T>();
+}
+
+describe("FlinkStatementResultsViewModel and FlinkStatementResultsManager", () => {
   let sandbox: sinon.SinonSandbox;
-  let manager: FlinkStatementResultsManager;
-  let flinkSqlStatementsApi: sinon.SinonStubbedInstance<StatementsSqlV1Api>;
-  let flinkSqlStatementResultsApi: sinon.SinonStubbedInstance<StatementResultsSqlV1Api>;
-  let mockSidecar: sinon.SinonStubbedInstance<sidecar.SidecarHandle>;
-  let mockStatement: FlinkStatement;
-  let refreshFlinkStatementStub: sinon.SinonStub;
-  let resourceLoader: CCloudResourceLoader;
+  let ctx: FlinkStatementResultsManagerTestContext;
   const expectedParsedResults = loadFixture(
     "flink-statement-results-processing/expected-parsed-results.json",
   );
+  let os: Scope;
+  let storage: WebviewStorage<ResultsViewerStorageState>;
+  let post: PostFunction;
+  let timestamp: Signal<number>;
+  let vm: FlinkStatementResultsViewModel;
 
-  beforeEach(() => {
+  beforeEach(async () => {
     sandbox = sinon.createSandbox();
-    // stub the sidecar getFlinkSqlStatementsApi API
-    mockSidecar = sandbox.createStubInstance(sidecar.SidecarHandle);
-    sandbox.stub(sidecar, "getSidecar").resolves(mockSidecar);
+    os = ObservableScope();
+    ctx = await createTestResultsManagerContext(sandbox, os);
 
-    flinkSqlStatementsApi = sandbox.createStubInstance(StatementsSqlV1Api);
-    mockSidecar.getFlinkSqlStatementsApi.returns(flinkSqlStatementsApi);
+    timestamp = os.produce(Date.now(), (ts, _signal) => {
```

Comment:
Remove unused `_signal` param. In javascript/typescript no need to declare params you're not going to reference.